New plugin - event bound images

jon-heard opened the pull request.

jon-heard committed:
New plugin - event bound images

seleb commented on src/event bound images.js:
```diff
@@ -0,0 +1,82 @@
+});
+
+// Update positions for all avatar-bound images, based on their events' positions.
+window.updateAllEventBoundImagePositions = () => {
```
why is this exposed globally? doesn't seem like you'd ever need to call it outside the plugin scope

seleb commented on src/event bound images.js:
```diff
@@ -0,0 +1,82 @@
+};
+
+// Update the position for a single avatar-bound image, based on its event's position.
+window.updateEventBoundImagePosition = (binding, currentRoomId) => {
```
same Q about global scope

seleb commented on src/event bound images.js:
```diff
@@ -0,0 +1,82 @@
+	}
+	// If there were no avatar-bound images up to now, start the frame updates.
+	if (!Object.keys(this.eventBoundImages).length) {
+		this.eventBoundImagesAnimationId = window.requestAnimationFrame(window.updateAllEventBoundImagePositions);
```
why are you updating the images via a separate raf loop? wouldn't it be much simpler to run this before/after bipsi's update/render calls?

there's also very little performance benefit to starting/stopping the raf loop based on whether there are any images to update since the update fn will return almost immediately when the bound images map is empty; it'd be simpler to just start the loop once and allow it to run indefinitely instead of managing the extra state

seleb commented on src/event bound images.js:
```diff
@@ -0,0 +1,82 @@
+// Update the position for a single avatar-bound image, based on its event's position.
+window.updateEventBoundImagePosition = (binding, currentRoomId) => {
+	if (window.roomFromEvent(window.PLAYBACK.data, binding.event).id !== currentRoomId) {
+		binding.image.x = 999;
```
this is hacky lol, i assume you're just trusting that 999 will be offscreen, but if that's the case:
- add a `// HACK: ` comment with that explanation
- use the actual bipsi map/tile size variables so it's guaranteed to be offscreen even if those change

jon-heard committed:
Added function "BIND_IMAGE_TO_EVENT" to match "SHOW_IMAGE"
Replaced custom RAF function with wrap of BipsiPlayback.render
Replaced global functions with ties into PLAYBACK
Replaced arbitrary offscreen image position to one based on ROOM_PX

jon-heard committed:
Tweaked a comment

jon-heard commented:
I've addressed your issues with this plugin.  It's got some other improvements too.

jon-heard committed:
added some missing function names

seleb committed:
add eslint globals

seleb committed:
fix typo

seleb commented:
:tada: This PR is included in version 4.8.0 :tada:

The release is available on [GitHub release](https://github.com/seleb/bipsi-hacks/releases/tag/v4.8.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
